lisp/emacs-lisp/emacs-ng.el

```diff
@@ -1 +1,112 @@
+(defvar straight-recipes-gnu-elpa-url)
+(defvar straight-repository-user)
+(defvar straight-repository-branch)
+(declare-function straight--reset-caches "straight")
+(declare-function straight-use-package-mode "straight")
+(declare-function straight-package-neutering-mode "straight")
+(declare-function straight-symlink-emulation-mode "straight")
+(declare-function straight-watcher-start "straight")
+(declare-function straight-live-modifications-mode "straight")
+(declare-function straight--modifications "straight")
+
+;;; Customization options
+
+;;;###autoload
+(defcustom ng/straight-bootstrap-at-startup t
+  "Whether to bootstrap straight.el when Emacs starts.
+If non-nil, straight.el is bootstrapped before reading the init
+file (but after reading the early init file).  This means that if
+you wish to set this variable, you must do so in the early init
+file.  Regardless of the value of this variable, straight.el recipes
+are not made available if `user-init-file' is nil (e.g. Emacs
+was started with \"-q\").
+
+Even if the value is nil, you can type \\[ng/bootstrap-straight] to
+bootstrap straight.el at any time, or you can
+call (ng/bootstrap-straight) in your init-file."
+  :type 'boolean
+  :group 'emacs-ng
+  :version "28")
+
+;;;###autoload
+(defvar ng/straight--bootstrapped nil
+  "Non-nil if `ng/bootstrap-straight' has been run.")
+
+;;;###autoload
+(defun ng/bootstrap-straight ()
+  "Bootstrap straight.el."
+
+  (interactive)
+
+  (require 'straight)
+
+  (setq ng/straight--bootstrapped t)
+
+  (straight--reset-caches)
+  (setq straight-recipe-repositories nil)
+
+  (straight-use-recipes '(org-elpa :local-repo nil))
+
+  (straight-use-recipes '(melpa :type git :host github
+                                :repo "melpa/melpa"
+                                :build nil))
+
+  (if straight-recipes-gnu-elpa-use-mirror
+      (straight-use-recipes
+       '(gnu-elpa-mirror :type git :host github
+                         :repo "emacs-straight/gnu-elpa-mirror"
+                         :build nil))
+    (straight-use-recipes `(gnu-elpa :type git
+                                     :repo ,straight-recipes-gnu-elpa-url
+                                     :local-repo "elpa"
+                                     :build nil)))
+
+  (straight-use-recipes '(el-get :type git :host github
+                                 :repo "dimitri/el-get"
+                                 :build nil))
+
+  (if straight-recipes-emacsmirror-use-mirror
+      (straight-use-recipes
+       '(emacsmirror-mirror :type git :host github
+                            :repo "emacs-straight/emacsmirror-mirror"
+                            :build nil))
+    (straight-use-recipes '(emacsmirror :type git :host github
+                                        :repo "emacsmirror/epkgs"
+                                        :nonrecursive t
+                                        :build nil)))
+
+  (straight-use-package `(straight :type git :host github
```

Comment:
Should we left this out?

Reply:
Since we use a submodule for straight, I guess we don't need this.